src/GregClient/AuthProviders/IAuthProvider.cs

```diff
@@ -36,4 +36,11 @@ public interface IOAuth2AuthProvider : IAuthProvider
         /// <param name="client"></param>
         new void SignRequest(ref RestRequest m, RestClient client);
     }
+    /// <summary>
+    /// Implement this interface to provide direct access to the current bearer token. 
+    /// </summary>
+    public interface IOAuth2AccessTokenProvider
```

Comment:
Would it help if this derives from `IOAuth2AuthProvider`?

Reply:
I don't think there is a big difference - I guess this is slightly more flexible, but it would be nice to break `IOAuth2AuthProvider` and merge these together in the near future - maybe at GL 25.